tests/gtests/internals/sdpa_internal.hpp

```diff
@@ -0,0 +1,182 @@
+            : primitive_desc(aengine, query_desc, key_desc, value_desc, nullptr,
+                    memory::data_type::undef, output_desc, false, 1, false,
+                    attr, kq_attr, vs_attr) {}
+
+        /// Constructs a primitive descriptor for a sdpa primitive
+        ///
+        /// @param aengine Engine to use.
+        /// @param query_desc Memory descriptor for query tensor.
+        /// @param key_desc Memory descriptor for key tensor.
+        /// @param value_desc Memory descriptor for value tensor.
+        /// @param output_desc Memory descriptor for output tensor.
+        /// @param attn_mask_desc Memory descriptor for attention mask tensor.
+        /// @param attr Primitive attributes to use. Attributes are optional
+        ///     and default to empty attributes.
+        primitive_desc(const engine &aengine, const memory::desc &query_desc,
+                const memory::desc &key_desc, const memory::desc &value_desc,
+                const memory::desc &attn_mask_desc,
+                const memory::desc &output_desc,
+                const primitive_attr &attr = default_attr(),
+                const primitive_attr &kq_attr = default_attr(),
+                const primitive_attr &vs_attr = default_attr())
+            : primitive_desc(aengine, query_desc, key_desc, value_desc,
+                    &attn_mask_desc, memory::data_type::undef, output_desc,
+                    false, 1, false, attr, kq_attr, vs_attr) {}
+
+        /// Constructs a primitive descriptor for a sdpa primitive from a C
+        /// API primitive descriptor that must have a matching kind.
+        ///
+        /// @param pd C API primitive descriptor for a sdpa primitive.
+        primitive_desc(dnnl_primitive_desc_t pd)
+            : dnnl::primitive_desc(pd, dnnl::primitive::kind::undef) {}
+
+        /// @copydoc dnnl::primitive_desc_base::src_desc()const
+        memory::desc query_desc() const {
+            return query_md((dnnl::query)query::src_md, 0);
+        }
+
+        /// @copydoc dnnl::primitive_desc_base::weights_desc()const
+        memory::desc key_desc() const {
+            return query_md((dnnl::query)query::src_md, 1);
+        }
+
+        /// @copydoc dnnl::primitive_desc_base::weights_desc()const
+        memory::desc value_desc() const {
+            return query_md((dnnl::query)query::src_md, 2);
+        }
+
+        /// @copydoc dnnl::primitive_desc_base::weights_desc()const
+        memory::desc attn_mask_desc() const {
+            return query_md((dnnl::query)query::src_md, 3);
+        }
+
+        /// @copydoc dnnl::convolution_forward::primitive_desc::bias_desc()const
+        memory::desc bias_desc() const {
+            return query_md((dnnl::query)query::weights_md, 1);
+        }
+
+        ///// @copydoc dnnl::primitive_desc_base::dst_desc()const
+        memory::desc dst_desc() const {
+            return query_md((dnnl::query)query::dst_md, 0);
+        }
```

Comment:
Something tells me those queries are not used at all. If they are not, let's drop them.